tripal/tripal
# REVEAL: Adding the title Templates to the content type YAML

# Tripal 4 Core Dev Task

## Issue #1563

### Tripal Version: 4.alpha1-dev

## Description

Currently the default title for any content type is a mishmash of all the required fields. For example, analysis is `[TripalEntityType__label], [analysis_program_version], [analysis_software]`

This specifies the defaults from  the yaml file `tripal_chado/config/install/tripal.tripalentitytype_collection.default_chado.yml` - see issue #1563 for the "once upon a time" tale of this 😕

## Testing

  * Build a docker on this branch
  * Go to `/admin/structure/bio_data`
  * In the "Label" column click on analysis, i.e. `/admin/structure/bio_data/manage/bio_data_2`
  * In **Advanced Settings** you should see the title format and url format specified in the yaml file `tripal_chado/config/install/tripal.tripalentitytype_collection.default_chado.yml`

Previously, this was **[TripalEntityType__label], [analysis_program_version], [analysis_software]**, now it is:

![title](https://github.com/tripal/tripal/assets/8419404/b2f60772-04d7-403f-8b8c-8a6f86c0a392)

Previously, this was **[type]/[TripalEntity__entity_id]**, now it is (ha ha look at that typo, let's fix it! https://github.com/tripal/tripal/pull/1583/commits/1871730397b850e4a93cddb4f6fc2075e5866ad7):

![url](https://github.com/tripal/tripal/assets/8419404/b6e582e9-6e18-4d11-9c3a-8fc90bf4bdc3)
